src/compile/scale/type.ts

```diff
@@ -16,56 +16,56 @@ export type RangeType = 'continuous' | 'discrete' | 'flexible' | undefined;
  */
 // NOTE: CompassQL uses this method.
 export default function type(
-  specifiedType: ScaleType, type: Type, channel: Channel, timeUnit: TimeUnit, mark: Mark,
+  specifiedType: ScaleType, type: Type, channel: Channel, timeUnit: TimeUnit, binned: boolean, mark: Mark,
```

Comment:
Now we have so many parameters, making it hard to read.

Following https://basarat.gitbooks.io/typescript/docs/tips/functionParameters.html, I wonder if we should group parts from fieldDef like this `fieldDefParts: {bin?: boolean, timeUnit?: TimeUnit  type: Type}`.  This will make the method slightly easier to read plus eliminate many `undefined` in the test.